src/modules/manager/actions.js

```diff
@@ -25,21 +27,22 @@ export const beginProjectEditing = basicAction(PROJECT_EDITING.BEGIN);
 export const endProjectEditing = basicAction(PROJECT_EDITING.END);
 
 
-const fetchDropBoxTree = networkAction(() => (dispatch, getState) => ({
+export const fetchDropBoxTree = networkAction(() => (dispatch, getState) => ({
     types: FETCH_DROP_BOX_TREE,
+    check: (state) => state.services.dropBoxService
+        && !state.dropBox.isFetching
+        && (!state.dropBox.tree.length || state.dropBox.isInvalidated),
```

Comment:
What is this condition for?

Reply:
it's another pattern of forcing a drop box (or other state) reload - instead of calling a non-checked fetch action version, we mark the state as invalidated and then whenever `fetchIfNeeded` or whatever is called, it will re-perform the fetch since the state is marked as invalid.

Reply:
is this fine?